test/unit/aws/test_source_ami.py

```diff
@@ -1,1005 +1,96 @@
-        "ImageType": "machine",
-        "Public": True,
-        "OwnerId": "099720109477",
-        "PlatformDetails": "Linux/UNIX",
-        "UsageOperation": "RunInstances",
-        "State": "available",
-        "BlockDeviceMappings": [
-            {
-                "DeviceName": "/dev/sda1",
-                "Ebs": {
-                    "DeleteOnTermination": True,
-                    "SnapshotId": "snap-0a31b4b0ef89e280e",
-                    "VolumeSize": 8,
-                    "VolumeType": "gp2",
-                    "Encrypted": False
-                }
-            },
-            {
-                "DeviceName": "/dev/sdb",
-                "VirtualName": "ephemeral0"
-            },
-            {
-                "DeviceName": "/dev/sdc",
-                "VirtualName": "ephemeral1"
-            }
-        ],
-        "Description": "Canonical, Ubuntu, 20.04 LTS, amd64 focal image build on 2020-09-16",
-        "EnaSupport": True,
-        "Hypervisor": "xen",
-        "Name": "ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20200916",
-        "RootDeviceName": "/dev/sda1",
-        "RootDeviceType": "ebs",
-        "SriovNetSupport": "simple",
-        "VirtualizationType": "hvm",
-        "DeprecationTime": "2022-09-17T16:24:38.000Z"
-    }),
+
+class AmiSpec(NamedTuple):
+    date: str
+    id: str
+
+
+# The following is based on a dump of data returned from aws cli:
+# aws ec2 --profile exa_individual_mfa describe-images
+# --owners 099720109477
+# --filters Name=name,Values=ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-*
+
+AMI_SPECS = [
+    AmiSpec("2020-10-14", "ami-056114420b6ed624e"),
+    AmiSpec("2022-03-31", "ami-0d672276deff62a7b"),
+    AmiSpec("2022-06-28", "ami-0d203747b007677da"),
+    AmiSpec("2022-06-10", "ami-0c9354388bb36c088"),
 ]
 
 
-def test_find_source_ami_returns_latest_ami(test_config):
+@pytest.fixture(scope="session")
+def sample_amis():
+    def ami(spec: AmiSpec):
+        suffix = spec.date.replace("-", "")
+        return Ami({
+            "Architecture": "x86_64",
+            "CreationDate": f"{spec.date}T16:39:06.000Z",
+            "ImageId": spec.id,
+            "OwnerId": "099720109477",
+            "State": "available",
+            "Name": f"ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-{suffix}",
+        })
+
+    return [ami(spec) for spec in AMI_SPECS]
+
+
+@pytest.fixture(scope="session")
+def aws_mock(sample_amis):
+    def mock_list_amis(filters: list[dict[str, str | list[str]]]) -> list[Ami]:
+        def matches(ami: Ami) -> bool:
+            for f in filters:
+                key = f["Name"]
+                att = key.title().replace("-", "")
+                for crit in f["Values"]:
+                    regex = crit.replace("*", ".*")
+                    value = ami._aws_object[att]
+                    if not re.match(regex, value):
+                        return False
+            return True
```

Comment:
Honestly, I think the matcher impl in the mock became so complicated, that it needs it's own test.
Can't we return a static value like before?